central/networkgraph/flow/datastore/internal/store/postgres/cluster_impl.go

```diff
@@ -5,28 +5,42 @@ import (
 
 	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/stackrox/rox/central/networkgraph/flow/datastore/internal/store"
+	"github.com/stackrox/rox/pkg/sync"
 )
 
 // NewClusterStore returns a new ClusterStore instance using the provided rocksdb instance.
 func NewClusterStore(db *pgxpool.Pool) store.ClusterStore {
 	return &clusterStoreImpl{
-		db: db,
+		db:        db,
+		flowStore: make(map[string]store.FlowStore),
 	}
 }
 
 type clusterStoreImpl struct {
-	db *pgxpool.Pool
+	db        *pgxpool.Pool
+	lock      sync.Mutex
+	flowStore map[string]store.FlowStore
```

Comment:
This store is created on a per cluster basis, I presume to improve performance as that pre-dates me.  The methods to create and get the store were essentially the same where it would return a new store based on cluster.  So while not expensive, having multiple stores servicing the same cluster makes it difficult to prevent deadlocks because each path through the system. has access to a different copy of the store.  So using mutexes to prevent deadlocks became ineffective.  Doing it this way allows me to use a mutex to prevent deadlocks on the bulk delete calls.